src/wallet/wallet.cpp

```diff
@@ -1325,7 +1325,7 @@ CAmount CWalletTx::GetLockedCredit() const
         return 0;
 
     // Must wait until coinbase is safely deep enough in the chain before valuing it
-    if (IsCoinBase() && GetBlocksToMaturity() > 0)
+    if ((IsCoinBase() || IsCoinStake()) && GetBlocksToMaturity() > 0)
```

Comment:
nit: this shouldn't be needed,  as `GetBlocksToMaturity` returns `0` for non-coinbase / non-coinstake txes

Reply:
true, there are lot of places doing the same in the wallet file. We should create a single PR cleaning them all.